p4-16/spec/P4-16-spec.mdk

```diff
@@ -3758,6 +3770,13 @@ expressions are legal:
   are copied, including its elements and their validity bits,
   as well as `nextIndex`.
 
+- `hs.minSizeInBits()` is a compile-time constant value of type `int`
+  equal to `n * h.minSizeInBits()`, where `h` is the header type on which
+  the header stack `hs` is defined.
+
+- `hs.minSizeInBytes()` is a compile-time constant value of type `int`
```

Comment:
Is the size of a header stack a statically known constant? If not, I don't think we can make `hs.minSizeInBytes()` a compile-time constant.

Reply:
During compile-time, the header stack size and elementType are known.  Hence, the total size of the header stack is also known.

https://github.com/p4lang/p4c/blob/master/ir/type.def#L343

Reply:
Given that the sweet spot use case for minSizeInBits/Bytes is for a single simple header, I am happy to remove it from the spec for everything else if no one minds.  i.e. do not define it for header stacks, and do not define it for header_union.